File: ArmstrongNumbers/armstrong.py
# Armstrong Numbers solution in Python 2 for CodeEval.com by Steven A. Dunn

def isArmstrong(N):
  sum = 0
  numDigits = len(N)
  for c in N:
    sum += (int(c) ** numDigits) 
  if sum == int(N): return True
  return False

import sys

for line in open(sys.argv[1], 'r'):
  if isArmstrong(line.rstrip('\n')): print True
  else: print False

File: ArmstrongNumbers/armstrong.rb
# armstrong numbers soln in rb for codeeval.com by steven a dunn

def isArmstrong(n)
    sum = 0
    numDigits = n.length

    n.split("").each do |c|
        sum += (c.to_i ** numDigits)
    end
    if sum == n.to_i
        return true
    end
    return false
end

File.open(ARGV[0], 'r').each_line do |line|
    if isArmstrong(line.chomp)
        puts "True"
    else
        puts "False"
    end
end